api/api/templates/admin/base_site.html

```diff
@@ -0,0 +1,22 @@
+{% extends "admin/base_site.html" %}
+
+{% block messages %}
+    {% if ENVIRONMENT != 'production' %}
+    <script>
+        function calculateDaysLeft() {
+            let dateToday = new Date();
+            let lastDayOfMonth = new Date(dateToday.getFullYear(), dateToday.getMonth()+1, 0).getDate();
+            let daysUntilEndOfMonth = lastDayOfMonth - dateToday.getDate();
+            document.getElementById("staging-db-refresh-days").innerText = daysUntilEndOfMonth;
+        }
+        document.addEventListener("DOMContentLoaded", calculateDaysLeft);
+    </script>
```

Comment:
The only concern here would be that this uses the local time of the browser client. it could be off by a day if it's, say, close to the end of the day in your timezone but UTC/the airflow server time (I'm not sure what timezone it's in, actually) has already moved over to the next day.

it's a bit more convoluted, but you could use UTC for this code like this:

```suggestion
    <script>
        function calculateDaysLeft() {
            let dateToday = new Date(Date.UTC(new Date().getUTCFullYear(), new Date().getUTCMonth(), new Date().getUTCDate()));
            let lastDayOfMonth = new Date(Date.UTC(dateToday.getUTCFullYear(), dateToday.getUTCMonth() + 1, 0)).getUTCDate();
            let daysUntilEndOfMonth = lastDayOfMonth - dateToday.getUTCDate();
            document.getElementById("staging-db-refresh-days").innerText = daysUntilEndOfMonth;
        }
        document.addEventListener("DOMContentLoaded", calculateDaysLeft);
    </script>
```

Reply:
Oh shoot, that's definitely an oversight - thank you!